src/paths.cc

```diff
@@ -37,7 +37,7 @@ int remove_directory_helper(const char *path) {
     } else {
         res = DeleteFile(path);
     }
-    if (res != 0) {
+    if (res == 0) {
```

Comment:
I had to double-check, and yes, DeleteFile and RemoveDirectory *both* return non-zero upon success.